### PR TITLE
Added non-null annotations

### PR DESCRIPTION
**Pull request by Nicholas-Baron**
Need to mark a lot of columns as non-null (thanks, Java).

### CODE REVIEW

File: rec_center_model/src/model/entities/ProfessorStudent.java [hunk 1]
```diff
@@ -9,19 +9,55 @@
...
 	protected Date enterDate;
 
-	@Column(name = "grad_date")
+	@Column(name = "grad_date", nullable = false)
 	protected Date gradDate;
 
+	@Column(nullable = false)
 	private String major;
 
+	@Column(nullable = false)
```
Comment (clcannon) [inline rec_center_model/src/model/entities/ProfessorStudent.java:21]: Shouldnt this be nullable? Not everybody has a minor.

File: rec_center_model/src/model/entities/Student.java [hunk 2]
```diff
@@ -9,16 +9,37 @@
...
 	protected Date enterDate;
 
-	@Column(name = "grad_date")
+	@Column(name = "grad_date", nullable = false)
 	protected Date gradDate;
 
+	@Column(nullable = false)
 	private String major;
 
+	@Column(nullable = false)
```
Comment (clcannon) [inline rec_center_model/src/model/entities/Student.java:21]: Same comment as professor/student, should minor be nullable?

**Review by clcannon — COMMENTED** [review]
Let me know what you think about the minor fields being nullable; its not critical so if its fine we can go ahead and merge it. Up to you

File: rec_center_model/src/model/entities/ProfessorStudent.java (hunk 1 above)
Comment (Nicholas-Baron) [reply rec_center_model/src/model/entities/ProfessorStudent.java:21]: What is the difference between the empty string and null here?
We could just as well use `""` to mean "no minor".

Comment (clcannon) [reply rec_center_model/src/model/entities/ProfessorStudent.java:21]: Cool I’m convinced! Approved

**Review by clcannon — APPROVED** [review]

**Pull request merged**